client/src/components/UserTrack/UserTrack.jsx

```diff
@@ -71,7 +71,7 @@ function UserTrack ({
         </div>
       </div>
       <div className="similarity">
-        <span>{similarityScore.toFixed(2)}</span>
+        <span>{similarityScore?.toFixed(2)}</span>
```

Comment:
What happens if it's undefined here?

Reply:
Oh woops, it should never be undefined. I forgot to remove it after I was done debugging.